test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go

```diff
@@ -193,6 +194,50 @@ var _ = Describe("Mixedcpus", Ordered, func() {
 				Expect(shared.Equals(*ppShared)).To(BeTrue(), "OPENSHIFT_SHARED_CPUS value not equal to what configure in the performance profile."+
 					"OPENSHIFT_SHARED_CPUS=%s spec.cpu.shared=%s", shared.String(), ppShared.String())
 			})
+			It("should contains the shared cpus after Kubelet restarts", func() {
+				rl := &corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("1"),
+					corev1.ResourceMemory: resource.MustParse("100Mi"),
+					sharedCpusResource:    resource.MustParse("1"),
+				}
+				p, err := createPod(ctx, testclient.Client, testutils.NamespaceTesting,
+					withRequests(rl),
+					withLimits(rl),
+					withRuntime(components.GetComponentName(profile.Name, components.ComponentNamePrefix)))
+				Expect(err).ToNot(HaveOccurred())
+				cfg := &controller.CpuSet{}
+				err = getter.Container(ctx, p, p.Spec.Containers[0].Name, cfg)
+				Expect(err).ToNot(HaveOccurred())
+				cgroupCpuSet, err := cpuset.Parse(cfg.Cpus)
+				Expect(err).ToNot(HaveOccurred())
+				shared, _ := cpuset.Parse(string(*profile.Spec.CPU.Shared))
+				Expect(cgroupCpuSet.Intersection(shared).List()).ToNot(BeEmpty(), "shared cpus are not in the pod cgroups; pod=%q, cgroupscpuset=%q sharedcpuset=%q",
+					fmt.Sprintf("%s/%s", p.Namespace, p.Name), cgroupCpuSet.String(), shared.String())
+
+				node := &corev1.Node{}
+				err = testclient.Client.Get(ctx, client.ObjectKey{Name: p.Spec.NodeName}, node)
+				Expect(err).ToNot(HaveOccurred())
+
+				cmd := kubeletRestartCmd()
+				// The command would fail since it aborts all the pods during restart
+				_, _ = nodes.ExecCommandOnNode(ctx, cmd, node)
+				// check that the node is ready after we restart Kubelet
+				nodes.WaitForReadyOrFail("post restart", node.Name, 20*time.Minute, 3*time.Second)
+
+				// giving kubelet more time to stabilize and initialize itself before
+				// moving on with the testing.
+				testlog.Infof("post restart: entering cooldown time: %v", restartCooldownTime)
+				time.Sleep(restartCooldownTime)
+				testlog.Infof("post restart: finished cooldown time: %v", restartCooldownTime)
```

Comment:
I understand why you do that, the problem this is inherently fragile and racy. I guess we can't do much better for the time being

Reply:
> but with the usual caveat: this part is ok, but we need tests which check the shared cpus from the perspective of the running container (vs execing into it) because running a container and execing into it can get different pictures. And it actually happened not so long ago.

Yes, this type of check should be replaced everywhere in the testing framework but it requires some extra work from my side.

Reply:
> I understand why you do that, the problem this is inherently fragile and racy. I guess we can't do much better for the time being

I Used the exact same logic from:
https://github.com/openshift/cluster-node-tuning-operator/blob/b185119c5d5d100a4a38342989e8aefa0e5428f6/test/e2e/performanceprofile/functests/9_reboot/devices.go#L212-L211

Reply:
yes, it's debatable even if I did it first :)